### PR TITLE
Optimize save queries

### PR DESCRIPTION
**Pull request by desoindx**

### CODE REVIEW

File: db/psychologists.js [hunk 1]
```diff
@@ -103,14 +118,17 @@ module.exports.savePsychologistInPG = async function savePsychologistInPG(psyLis
   const updatedAt = date.getDateNowPG(); // use to perform UPSERT in PG
   const universities = await dbUniversities.getUniversities();
 
+  const psychologists = await getPsychologistsByIds(psyList.map((psy) => psy.dossierNumber));
+  const psychologistsToInsert = [];
   const upsertArray = psyList.map(async (psy) => {
     psy.languages = addFrenchLanguageIfMissing(psy.languages);
     psy.assignedUniversityId = dbUniversities.getAssignedUniversityId(psy, universities);
```
Comment (sandrica89) [inline db/psychologists.js:125]: This can be done only if `!psyInDB` since we never update this field in other cases.

File: db/psychologists.js [hunk 2]
```diff
@@ -103,14 +118,17 @@ module.exports.savePsychologistInPG = async function savePsychologistInPG(psyLis
...
 
     try {
-      const psyInDb = await getPsychologistById(psy.dossierNumber);
+      const psyInDb = psychologists[psy.dossierNumber];
       if (!psyInDb) {
-        return knex(psychologistsTable).insert(psy);
+        psychologistsToInsert.push(psy);
+        return false;
```
Comment (sandrica89) [inline db/psychologists.js:131]: Why return false here?

File: db/psychologists.js [hunk 3]
```diff
@@ -136,7 +154,8 @@ module.exports.savePsychologistInPG = async function savePsychologistInPG(psyLis
     }
   });
 
-  const query = await Promise.all(upsertArray);
+  upsertArray.push(knex(psychologistsTable).insert(psychologistsToInsert));
+  const query = await Promise.all(upsertArray.filter((x) => x));
```
Comment (sandrica89) [inline db/psychologists.js:158]: I don't understand `.filter((x) => x)`

File: db/psychologists.js [hunk 4]
```diff
@@ -136,7 +154,8 @@ module.exports.savePsychologistInPG = async function savePsychologistInPG(psyLis
     }
   });
 
-  const query = await Promise.all(upsertArray);
+  upsertArray.push(knex(psychologistsTable).insert(psychologistsToInsert));
```
Comment (sandrica89) [inline db/psychologists.js:157]: What if the insert query fails? Before we had a try catch logging the error and resolving the promise. Here, the promise will end with an error, will we know by any mean that the insertions failed?

**Review by sandrica89 — CHANGES_REQUESTED** [review]
See PR comments

File: db/psychologists.js (hunk 4 above)
Comment (desoindx) [reply db/psychologists.js:157]: it fails properly 
image

**Review by sandrica89 — APPROVED** [review]

**Pull request merged**